pkg/gadget-service/service-oci.go

```diff
@@ -136,9 +136,15 @@ func (s *Service) RunGadget(runGadget api.GadgetManager_RunGadgetServer) error {
 		return fmt.Errorf("expected version to be %d, got %d", api.VersionGadgetRunProtocol, ociRequest.Version)
 	}
 
+	// Create payload buffer
+	outputBuffer := make(chan *api.GadgetEvent, s.eventBufferLength)
+
 	// Create a new logger that logs to gRPC and falls back to the standard logger when it failed to send the message
 	logger := logger.NewFromGenericLogger(&Logger{
-		send:           runGadget.Send,
+		send: func(event *api.GadgetEvent) error {
+			outputBuffer <- event
```

Comment:
This is again a bit dangerous - I wonder if we should add the severity to the callback to decide on what to do.

As it is right now, that line is blocking if the buffer is full - hence it might cause the caller to block and so on. Down in the message pump we have the `select`/`default` construct that makes it non-blocking; if the buffer is full, the default will be used. I think the "best" approach would be to do the select/default and then in the default log locally on the server as fallback.